projects/storefrontlib/src/cms-components/checkout/components/delivery-mode/delivery-mode.component.spec.ts

```diff
@@ -243,8 +249,9 @@ describe('DeliveryModeComponent', () => {
 
     it('should call "back" function after being clicked', () => {
       spyOn(component, 'back');
-      getContinueBtn().nativeElement.click();
+
       fixture.detectChanges();
+      getContinueBtn().nativeElement.click();
```

Comment:
you should detect changes after the click

Reply:
also, this test (`it()`) is about 'go back' button and not 'continue', right? not sure why you touched it

Reply:
Hi buddy, when I finish correction, the unit test for back button also fails, which makes me quite surprised. The failure is caused before the following line for BACK button unit test fails: getContinueBtn() function returns null, 

```TypeScript
getContinueBtn().nativeElement.click();
```

And I found out if I call **fixture.detectChanges();** before  **getContinueBtn().nativeElement.click();**, then the getContinueBtn can return the expected button. I am still investigating why.

Reply:
got confirmation from Kacper:

In order to make continue button accessible by code like below:

```typescript
fixture.debugElement.query(By.css('.cx-checkout-btns .btn-primary'));
```

we have to call fixture.detectChanges() first. 

Check test result from Kacper:

![image](https://user-images.githubusercontent.com/58975336/96075251-3aae6100-0edd-11eb-806e-cc9c0c9b96e7.png)

I attach our discussion log for reference:

![image](https://user-images.githubusercontent.com/58975336/96075312-59acf300-0edd-11eb-9689-d217dbd836b6.png)